code/modules/projectiles/guns/projectile/boltaction.dm

```diff
@@ -0,0 +1,93 @@
+/obj/item/weapon/gun/projectile/boltaction

```

Comment:
Use `/obj/item/weapon/gun/projectile/rifle/boltaction` instead. It already takes care of all of the wielding stuff, and I don't think there's a reason to do it otherwise.